server/models/dataset.py

```diff
@@ -104,4 +109,93 @@ def findOne(self, query=None, **kwargs):
     def validate(self, doc, **kwargs):
         # TODO: implement
         # raise ValidationException
+
+        # Validate name. This is redundant, because Folder also validates the
+        # name, but this allows for a more specific error message.
+        doc['name'] = doc['name'].strip()
+        if not doc['name']:
+            raise ValidationException('Dataset name must not be empty.', 'name')
         return Folder.validate(self, doc, **kwargs)
+
+    def processDataset(self, uploadFolder, user, name, description, license,
+            signature, anonymous, attribution):
+        """
+        Process an uploaded dataset. This upload folder is expected to contain a
+        .zip file of images and a .csv file that contains metadata about the
+        images. The images are extracted to a new folder in the "Phase 0"
+        collection and then processed.
+        """
+        if not uploadFolder:
+            raise ValidationException(
+                'No files were uploaded.', 'uploadFolder')
+
+        zipFileItems = [f for f in self.model('folder').childItems(uploadFolder)
+                        if mimetypes.guess_type(f['name'], strict=False)[0] in
+                        ZIP_FORMATS]
```

Comment:
`ZIP_FORMATS` is only used here. Go ahead and move it over to this file, out of `upload.py`.a